File: Python/0012.py
class Solution:
    def intToRoman(self, num: int) -> str:
        # Define Roman numeral symbols and their corresponding values
        symbols = ['M', 'CM', 'D', 'CD', 'C', 'XC', 'L', 'XL', 'X', 'IX', 'V', 'IV', 'I']
        values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        
        # Initialize an empty string to store the Roman numeral representation
        result = ''
        
        # Initialize index to iterate through symbols and values
        i = 0
        
        # Iterate through each symbol and value pair
        while num > 0:
            # While num is greater than or equal to the current value, append the corresponding symbol
            while num >= values[i]:
                result += symbols[i]
                num -= values[i]  # Subtract the value from num to reduce it
            i += 1  # Move to the next symbol
        
        return result  # Return the final Roman numeral representation

File: Scala/0012.scala
object Solution {
  def intToRoman(num: Int): String = {
    // Define the Roman numerals and their corresponding values
    val romanMap = List(
      (1000, "M"),
      (900, "CM"),
      (500, "D"),
      (400, "CD"),
      (100, "C"),
      (90, "XC"),
      (50, "L"),
      (40, "XL"),
      (10, "X"),
      (9, "IX"),
      (5, "V"),
      (4, "IV"),
      (1, "I")
    )

    // Initialize result as an empty string
    var result = new StringBuilder
    var remaining = num

    // Iterate through the list of values and symbols
    for ((value, symbol) <- romanMap) {
      // While the remaining number is greater than or equal to the value
      while (remaining >= value) {
        result.append(symbol)
        remaining -= value
      }
    }

    // Convert StringBuilder to String and return the result
    result.toString
  }
}
